lib/pbench/test/unit/server/query_apis/test_search.py

```diff
@@ -0,0 +1,183 @@
+import pytest
```

Comment:
I'd like to see some fully-validated test cases for different JSON through your `flatten_json` method. You can do that with additional API cases here, or work out a "real" unit test that acts directly on a `SearchAPI` class. I'd like to see tests for each of the top-level `if` cascade (`dict`, `list`, scalar), plus your "renaming" suffix code, at least. This is fairly convoluted code; not only should it be thoroughly tested, but detailed test examples will also help readers understand it.